Flagsmith.Client.Test/FlagsmithTest.cs

```diff
@@ -0,0 +1,193 @@
+﻿using Moq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Threading;
+using Xunit;
+namespace Flagsmith.FlagsmithClientTest
+{
+    public class FlagsmithTest
+    {
+        [Fact]
+        public void TestFlagsmithStartsPollingManagerOnInitIfEnabled()
+        {
+            var mockHttpClient = HttpMocker.MockHttpResponse(new HttpResponseMessage
+            {
+                StatusCode = System.Net.HttpStatusCode.OK,
+                Content = new StringContent(Fixtures.JsonObject.ToString())
+            });
+            var flagsmithClientTest = new FlagsmithClient(Fixtures.ApiKey, httpClient: mockHttpClient.Object, enableClientSideEvaluation: true);
+            mockHttpClient.verifyHttpRequest(HttpMethod.Get, "/api/v1/environment-document/", Times.Once);
+        }
+        [Fact]
+        public async void TestUpdateEnvironmentSetsEnvironment()
+        {
+            var ss = Times.Once;
+            var mockHttpClient = HttpMocker.MockHttpResponse(new HttpResponseMessage
+            {
+                StatusCode = System.Net.HttpStatusCode.OK,
+                Content = new StringContent(Fixtures.JsonObject.ToString())
+            });
+            var flagsmithClientTest = new FlagsmithClient(Fixtures.ApiKey, enableClientSideEvaluation: true, httpClient: mockHttpClient.Object);
+            mockHttpClient.verifyHttpRequest(HttpMethod.Get, "/api/v1/environment-document/", Times.Once);
+            await flagsmithClientTest.GetEnvironmentFlags();
+            mockHttpClient.verifyHttpRequest(HttpMethod.Get, "/api/v1/environment-document/", Times.Once);
```

Comment:
I think perhaps the following would be more useful here? Perhaps we can add both? 

```dotnet
mockHttpClient.verifyHttpRequest(HttpMethod.Get, "/api/v1/flags/", Times.Never);
```

Reply:
added